interface/forms/vitals/templates/vitals/vitals_textbox_conversion.tpl

```diff
@@ -0,0 +1,74 @@
+<!-- USA row comes first -->
+    {if $units_of_measurement == $MEASUREMENT_METRIC_ONLY}
+    <tr class="hide">
+    {else}
+    <tr>
+    {/if}
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_METRIC}
+        <td class="unfocus graph" id="{$input}">
+    {else}
+        <td class="graph" id="{$input}">
+    {/if}
+            {xlt t=$title}
+        </td>
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_METRIC}
+        <td class="unfocus">
+    {else}
+        <td>
+    {/if}
+            {xlt t=$unit}
+        </td>
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_METRIC}
+        <td class="valuesunfocus">
+    {else}
+        <td class='currentvalues p-2'>
+    {/if}
+            <input type="text" class="form-control" size='5' name='{$input}' id='{$input}_input'
+                   value="{if $vitals->$vitalsValue() != 0}{$vitals->$vitalsValue()|attr}{/if}"
+                   onChange="convUnit('usa', '{$unit}','{$input}_input')" title='{xla t=$vitalsValueUSAHelpTitle|default:''}'/>
+        </td>
+    <td class="editonly">
+        { include file='vitals_interpretation_selector.tpl' vitalDetails=$vitals->get_details_for_column($input) }
+    </td>
+    { include file='vitals_historical_values.tpl' useMetric=false vitalsValue=$vitalsValue vitalsValueMetric=$vitalsValueMetric
+            results=$results }
+    </tr>
+
+<!-- Metric row comes second -->
+{if $units_of_measurement == $MEASUREMENT_USA_ONLY}
+    <tr class="hide">
+{else}
+    <tr>
+{/if}
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_USA}
+        <td class="unfocus graph" id="{$input}_metric">
+    {else}
+        <td class="graph" id="{$input}_metric">
+    {/if}
+            {xlt t=$title}
+        </td>
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_USA}
+        <td class="unfocus">
+    {else}
+        <td>
+    {/if}
+            {xlt t=$unitMetric}
+    </td>
+    {if $units_of_measurement == $MEASUREMENT_PERSIST_IN_USA}
+        <td class="valuesunfocus">
+    {else}
+        <td class='currentvalues p-2'>
+    {/if}
+            <input type="text" class="form-control" size='5' id='{$input}_input_metric'
+                   value="{if $vitals->$vitalsValueMetric() != 0}{$vitals->$vitalsValueMetric()|attr}{/if}"
+                   onChange="convUnit('metric', '{$unit}','{$input}_input')"/>
```

Comment:
```
onChange="convUnit('metric', {$unit|attr_js},'{$input|attr}_input')"/>
```

Reply:
So the $unit|attr_js breaks giving me a value of "&quot;kg&quot;" which fails the javascript.  If I use $unit|js_escape it throws in an extra space.  $unit is actually a defined hard-coded value in the template inside interface/forms/vitals/templates/vitals/general_new.html.  Does it really need to escape as its not coming from user input at all.

Reply:
So I escape the $input but I'm leaving the $unit alone.

Reply:
yep, pretty much escape anything. otherwise, will have nightmares of cross-scripting attacks having access to the cookie with session id (since openemr needs cookie_httponly set to false) :)

![tenor (5)](https://user-images.githubusercontent.com/278968/126744400-84a7c22f-c145-4665-bc22-8c5ac0428ea1.gif)